coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -4226,27 +4251,31 @@ def gelu(context, node):
     context.add(res)
 
 
-@register_torch_op(torch_alias=["slice"])
-def _slice(context, node):
-    inputs = _get_inputs(context, node, expected=5)
+@register_torch_op(torch_alias=["slice_copy.tensor"])
+def slice(context, node):
+    inputs = _get_inputs(context, node, min_expected=1)
     x = inputs[0]
-    dim = inputs[1].val
+    dim = 0 if len(inputs) < 2 else inputs[1].val
 
-    if inputs[2] and inputs[2].val is not None:
-        start = inputs[2].val
-    elif isinstance(inputs[2], Var):
-        start = inputs[2]
-    else:
-        start = 0
-
-    if inputs[3] and inputs[3].val is not None:
-        end = inputs[3].val
-    elif isinstance(inputs[3], Var):
-        end = inputs[3]
-    else:
-        end = None
+    start = 0
+    if len(inputs) > 2 and inputs[2] is not None:
+        start = inputs[2].val if inputs[2].val is not None else inputs[2]
+
+    end = None
+    if len(inputs) > 3 and inputs[3] is not None:
+        if inputs[3].val is not None:
+            end = inputs[3].val
+            if end < 0:
```

Comment:
why are we speacial handling `end` but not `start`?
I guess both of them can be negative?
Do we have test for these?

Reply:
Turns out we don't 🤣 we only test dynamic `slice`, but not regular static `slice`

Added the negative `start` handling, as well as a static `slice` unit test

Reply:
I think what I meant is,
in MIL `slice_by_index` already handles negative indexing (`start / end`),
so why do we still need to special handling by adding `x.shape[0]` to make the value positive?
Your changes look good to me, I am just wondering if those negative cases exposed some "bugs that we ignored"?

Reply:
Replied in another comment, copying it here:

TLDR: Actually, the error was fixed, no need to treat negative indices now 👍 removed these negative indices treatments

## Context
I used to encounter an error from out-of-range index:
* PyTorch use `INT64_MAX` as default `end`, in which case the last element is included in the slice result
* We use `int32` during conversion, so `INT64_MAX` becomes `-1`
* This `-1` confuses `mb.slice_by_index`, excluding the last element from the slice result
* This is not an issue anymore, since I revised `mb.const` for `int64`, to clip out-of-range value to `int32` max (or min)